### ISSUE TITLE
Bug: (fill in)

### ISSUE THREAD

**Issue by luckylooke** [eslint/eslint#16238]
### Environment

Node version: v16.15.1
npm version: v8.11.0
Local ESLint version: 8.22.0
Global ESLint version: none
Operating System: MacOS Monterey 12.5.1

### What parser are you using?

vue-eslint-parser

### What did you do?

<details>
<summary>Configuration</summary>

eslintrc.mjs
```js
/* eslint-env node */
import "@rushstack/eslint-patch/modern-module-resolution";

module.exports = {
  root: true,
  extends: [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript/recommended",
    "@vue/eslint-config-prettier",
  ],
  rules: {
    "no-console": "error",
    "@typescript-eslint/no-explicit-any": "off",
    quotes: [
      "warn",
      "single",
      { avoidEscape: true, allowTemplateLiterals: true },
    ],
  },
  env: {
    "vue/setup-compiler-macros": true,
  },
  ignorePatterns: ["dist/*"], // because dist is not in .gitignore anymore
};
```
</details>

```js
// example.js

export const xxx = "xxx";
```

### What did you expect to happen?

quotes to be autofixed

### What actually happened?

it is not fixed whatever I tried

Running
`./node_modules/.bin/eslint ./example.js --fix --debug`

it says that it is applying fixes, but files did not changed
```js

... lots of debug info ...

  eslint:source-code-fixer Applying fixes +7ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/laky/Projects/election-vue/example.js (pass 9) +1ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Parsing: /Users/laky/Projects/election-vue/example.js +4ms
  eslint:linter Parsing successful: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Scope analysis: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Scope analysis successful: /Users/laky/Projects/election-vue/example.js +1ms
  eslint:linter Generating fixed text for /Users/laky/Projects/election-vue/example.js (pass 9) +7ms
  eslint:source-code-fixer Applying fixes +13ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Linting code for /Users/laky/Projects/election-vue/example.js (pass 10) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Parsing: /Users/laky/Projects/election-vue/example.js +1ms
  eslint:linter Parsing successful: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Scope analysis: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Scope analysis successful: /Users/laky/Projects/election-vue/example.js +1ms
  eslint:linter Generating fixed text for /Users/laky/Projects/election-vue/example.js (pass 10) +5ms
  eslint:source-code-fixer Applying fixes +7ms
  eslint:source-code-fixer Found fixes to apply +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Parsing: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Parsing successful: /Users/laky/Projects/election-vue/example.js +1ms
  eslint:linter Scope analysis: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:linter Scope analysis successful: /Users/laky/Projects/election-vue/example.js +0ms
  eslint:file-enumerator Complete iterating files: ["./example.js"] +808ms
  eslint:cli-engine Linting complete in: 809ms +217ms
  eslint:cli Fix mode enabled - applying fixes +824ms

/Users/laky/Projects/election-vue/example.js
  1:20  warning  Strings must use singlequote  quotes

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by luckylooke** [comment]
It is something related to configuration, as I have tried to replace content of eslintrc.mjs by this:

```js
/* eslint-env node */
module.exports = {
  parser: '@typescript-eslint/parser',
  rules: {
    quotes: [
      "warn",
      "single",
      { avoidEscape: true, allowTemplateLiterals: true },
    ],
  },
};
```

and it works as expected 🤔

**Comment by luckylooke** [comment]
I have debugged that "@vue/eslint-config-prettier" is the cause of the issue.
```js
extends: [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript/recommended",
    // "@vue/eslint-config-prettier",
  ],
  ```
  once I comment it out it works as expected
  
  So I will open issue in their repo.
